File: ios/Sources/SurveySparrowIonicPlugin2Plugin/SurveySparrowIonicPlugin2Plugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SurveySparrowIonicPlugin2Plugin)
public class SurveySparrowIonicPlugin2Plugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "SurveySparrowIonicPlugin2Plugin"
    public let jsName = "SurveySparrowIonicPlugin2"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = SurveySparrowIonicPlugin2()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/surveysparrow/plugins/ionic2/SurveySparrowIonicPlugin2Plugin.java
package com.surveysparrow.plugins.ionic2;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SurveySparrowIonicPlugin2")
public class SurveySparrowIonicPlugin2Plugin extends Plugin {

    private SurveySparrowIonicPlugin2 implementation = new SurveySparrowIonicPlugin2();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}
